testing/matrix-sdk-test/src/test_json/keys_query_sets.rs

```diff
@@ -45,246 +316,108 @@ impl KeyDistributionTestData {
+    }
+
     /// Current user keys query response containing the cross-signing keys
     pub fn me_keys_query_response() -> KeyQueryResponse {
-        let data = json!({
-            "master_keys": {
-                "@me:localhost": {
-                    "keys": {
-                        "ed25519:KOS8zz9SJnMOxpfPOx9LO2+abuEcnZP/lxDo5RsXao4": "KOS8zz9SJnMOxpfPOx9LO2+abuEcnZP/lxDo5RsXao4"
-                    },
-                    "signatures": {
-                        "@me:localhost": {
-                            "ed25519:KOS8zz9SJnMOxpfPOx9LO2+abuEcnZP/lxDo5RsXao4": "5G9+Ns28rzNd+2DvP73Y0orr8sxduRQcrJj0YB7ZygH7oeXshvGLeQn6mcNs7q7ZrMR5bYlXxopufKSWWoKpCg",
-                            "ed25519:YVKUSVBKWX": "ih1Kmj4dTB1AjjkwrLA2qIL3e/oPUFisP5Ic8kGp29wrpoHokasKKnkRl1zS7zq6iBcOL6aOZLPPX/ZHYCX5BQ"
-                        }
-                    },
-                    "usage": [
-                        "master"
-                    ],
-                    "user_id": "@me:localhost"
-                }
-            },
-            "self_signing_keys": {
-                "@me:localhost": {
-                    "keys": {
-                        "ed25519:9gXJQzvqZ+KQunfBTd0g9AkrulwEeFfspyWTSQFqqrw": "9gXJQzvqZ+KQunfBTd0g9AkrulwEeFfspyWTSQFqqrw"
-                    },
-                    "signatures": {
-                        "@me:localhost": {
-                            "ed25519:KOS8zz9SJnMOxpfPOx9LO2+abuEcnZP/lxDo5RsXao4": "amiKDLpWIwUQPzq+eov6KJsoskkWA1YzrGNb7HF3OcGV0nm4t7df0tUdZB/OpREtT5D78BKtzOPUipde2DxUAw"
-                        }
-                    },
-                    "usage": [
-                        "self_signing"
-                    ],
-                    "user_id": "@me:localhost"
-                }
-            },
-            "user_signing_keys": {
-                "@me:localhost": {
-                    "keys": {
-                        "ed25519:mvzOc2EuHoVfZTk1hX3y0hyjUs4MrfPv2V/PUFzMQJY": "mvzOc2EuHoVfZTk1hX3y0hyjUs4MrfPv2V/PUFzMQJY"
-                    },
-                    "signatures": {
-                        "@me:localhost": {
-                            "ed25519:KOS8zz9SJnMOxpfPOx9LO2+abuEcnZP/lxDo5RsXao4": "Cv56vTHAzRkvdcELleOlhECZQP0pXcikCdEZrnXbkjXQ/k0ZvVOJ1beG/SiH8xc6zh1bCIMYv96C9p8o+7VZCQ"
-                        }
-                    },
-                    "usage": [
-                        "user_signing"
-                    ],
-                    "user_id": "@me:localhost"
-                }
-            }
+        let builder = KeyQueryResponseTemplate::new(Self::me_id().to_owned())
+            .with_cross_signing_keys(
+                Ed25519SecretKey::from_base64(Self::MASTER_KEY_PRIVATE_EXPORT).unwrap(),
+                Ed25519SecretKey::from_base64(Self::SELF_SIGNING_KEY_PRIVATE_EXPORT).unwrap(),
+                Self::me_private_user_signing_key(),
+            );
+
+        let response = builder.build_response();
+        with_settings!({sort_maps => true}, {
```

Comment:
What happened with the idea of putting the `assert_json_snapshot()` into the `build_response()` command?

Reply:
Well, I wanted to start with some snapshots before I added `build_response`, so that I could check what I was changing.

We could probably move the snapshot now, but it's a whole other line of investigation. `panic::Location` doesn't really work, because it ties the name of the snapshot to the *line number* where `build_response` was called, which is obviously not stable. I don't know if we could do something with [backtrace](https://docs.rs/backtrace/latest/backtrace/index.html), but I'm inclined to leave it for now.